### PR TITLE
Valid_move? for Rook

### PR DESCRIPTION
**Pull request by mumtaz7**
Hello, 

I create the valid_move? method for the rook piece and also the specs, which allow the rook to move both horizontally and vertically, but do not allow it to move diagonally. There is a merge conflict in the factories file, which I am fixing now. All tests are passing.

### CODE REVIEW

File: app/models/rook.rb [hunk 1]
```diff
@@ -1,2 +1,19 @@
...
+    y_difference = y_coordinate - destination_y
+
+    x_difference.zero? && y_difference != 0
+  end
+
+  def horizontal?(destination_x, destination_y)
+    x_difference = x_coordinate - destination_x
+    y_difference = y_coordinate - destination_y
+
+    x_difference != 0 && y_difference.zero?
```
Comment (sylyu) [inline app/models/rook.rb]: mixed != and .zero?  should use .nonzero?

File: spec/factories/pieces.rb [hunk 2]
```diff
@@ -8,31 +8,46 @@
     icon ''
     color 'white'
 
-    factory :pawn, class: Pawn, parent: :piece do
+    factory :pawn, class: Pawn do
```
Comment (sylyu) [inline spec/factories/pieces.rb]: does this work?  cool!  I thought we had to inherit from parent.

Comment (mumtaz7) [inline spec/factories/pieces.rb]: yes it worked for me! celia helped me out with this. didn't seem to give me any issues :)

File: app/models/rook.rb (hunk 1 above)
Comment (mumtaz7) [inline app/models/rook.rb]: ah good point! thank you :) i'll change that

File: app/models/rook.rb [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+    x_difference = x_coordinate - destination_x
+    y_difference = y_coordinate - destination_y
+
+    x_difference.nonzero? && y_difference.zero?
+  end
+
+  def valid_move?(x, y)
+    vertical?(x, y) && !obstructed?(x, y) || horizontal?(x, y) && !obstructed?(x, y)
+  end
 end
```
Comment (mscoutermarsh) [inline app/models/rook.rb]: Nice job!

The `horizontal?` and `vertical?` methods could be made private. Since I think they are only used in this class.

Comment (mumtaz7) [inline app/models/rook.rb]: thank you, mike!! :) ok good to know!

**Pull request merged**